src/react-super-select.js

```diff
@@ -295,7 +295,8 @@ class ReactSuperSelect extends React.Component {
            aria-controls={this._ariaGetListId()}
            aria-label={placeholderString}
            aria-multiselectable={this._isMultiSelect()}
-           tabIndex="0">
+           tabIndex="0"
+           {...this.props.inputProps}>
```

Comment:
this needs to be destructured or put first in the component so that overriding props which would break the component's regular behavior cannot occur.

Reply:
The reason why it's the last prop is to be able to override `aria-label` (see my use case in the issue description). But you're right, there probably is no reason to override any other of those props. I could move `inputProps` up as the first prop and add a special behavior for `aria-label`, e.g.:

```
aria-label={(inputProps && inputProps["aria-label"]) || placeholderString}
```

One could also add a top level optional `aria-label` prop instead, leading to `aria-label={this.props["aria-label"] || placeholderString}`.

What do you think about this?

Reply:
You should be able to use ES6 destructuring to filter out the props which need to be protected.

```
// Use ES6 destructuring to disallow overriding props the control needs to own
const {
    className,
    onBlur,
    onKeyDown,
    role,
    'aria-activedescendant': ariaActivedescendant,
    'aria-disabled': ariaDisabled,
    'aria-haspopup': ariaHaspopup,
    'aria-controls': ariaControls,
    'aria-multiselectable': ariaMultiselectable,
    tabIndex,
    ...overridableInputProps
}

<div ref={(c) => {this._rssDOM.triggerDiv = c }}
   className={triggerClasses}
   onBlur={this.props.onBlur}
   onClick={this.toggleDropdown}
   onKeyDown={this._handleKeyDown}
   role="combobox"
   aria-activedescendant={this._ariaGetActiveDescendentId()}
   aria-disabled={this.props.disabled}
   aria-haspopup={true}
   aria-controls={this._ariaGetListId()}
   aria-label={placeholderString}
   aria-multiselectable={this._isMultiSelect()}
   tabIndex="0"
    {...overridableInputProps}
>
```

Reply:
Using destructuring would introduce unnecessary/unused variables, so I solved this with `_.omit` instead. I've also added a test to make sure protected props are not overwritten.

Reply:
That is fine, There may be some utility in allowing the className to pass through?  Thoughts?

Reply:
Makes sense, I've added this as well :smile: